### PR TITLE
feat: Allow buttons to have fully rounded/circular corners

### PR DESCRIPTION
**Pull request by gnapse**
## Short description

See https://twist.com/a/1585/ch/81455/t/3592631/

## PR Checklist

-   [ ] Added tests for bugs / new features
-   [x] Updated docs (storybooks, readme)
-   [x] Executed `npm run validate` and made sure no errors / warnings were shown
-   [x] Described changes in `CHANGELOG.md`
-   [ ] Bumped version in `package.json` and `package-lock.json` (`npm --no-git-tag-version version <major|minor|patch>`) [ref](https://docs.npmjs.com/cli/v6/commands/npm-version)
-   [ ] Updated all static build artifacts (`npm run build-all`)

## Versioning

New feature. New minor version.

### CODE REVIEW

**Comment by Bloomca** [comment]
@gnapse sorry I don't fully get it -- is it ready for review, or is it still a draft?

**Comment by gnapse** [comment]
> @gnapse sorry I don't fully get it -- is it ready for review, or is it still a draft?

Sorry for the confusion. I left it as draft while I went ahead and created the Twist thread. Since I was asking there about how to design the API, I was not sure if it would change. But the essence remains, regardless of how the prop and its value are designed.

I'll remove the draft status now, since it appears that the preferred approach is the one I already used here (`shape="rounded"`).

File: src/new-components/base-button/base-button.module.css [hunk 1]
```diff
@@ -119,6 +119,10 @@
     --reactist-spinner-fill: var(--reactist-btn-idle-fill);
 }
 
+.baseButton.shape-rounded {
+    border-radius: 1000px;
```
Comment (Bloomca) [inline src/new-components/base-button/base-button.module.css:123]: it doesn't change anything, just curious why not to use `border-radius: 100%`?

**Review by Bloomca — APPROVED** [review]
Code looks good.

I started the storybook and I didn't find any examples of how rounded buttons would look like (the property is documented though). Do you think it is worth to add a section with examples? I imagine a regular rounded button, and then an icon to show that it becomes round.

File: src/new-components/base-button/base-button.module.css (hunk 1 above)
Comment (gnapse) [reply src/new-components/base-button/base-button.module.css:123]: I've never understood why, but that has an unintended effect:

CleanShot 2022-07-04 at 20 05 28@2x

I've always had to use a very large number when wanting a full round border radius.

Comment (frankieyan) [reply src/new-components/base-button/base-button.module.css:123]: We could reserve this one for `shape="fish"`?

**Pull request merged**